src/common/flowTypes.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+import { ITEM_TYPE_FOLDER, ITEM_TYPE_FILE, ITEM_TYPE_WEBLINK } from './constants';
+
+type InlineNoticeType = 'warning' | 'error' | 'success' | 'info';
+type ItemType = ITEM_TYPE_FOLDER | ITEM_TYPE_FILE | ITEM_TYPE_WEBLINK;
```

Comment:
Should these be typeof ITEM_TYPE_FOLDER | typeof ITEM_TYPE_FILE | typeof ITEM_TYPE_WEBLINK?
And then in the constants file:

ITEM_TYPE_FOLDER: 'folder' = 'folder';
...

off topic: this is another syntax I learnt recently
https://flow.org/try/#0MYewdgzgLgBAygSQFoFEYF4YG8BQN-wCyAggDKkBcMAFAOQQC2AhgDYu1X3Nu0CUANHgKEUAEQQBVQlToMApgBMAlgFcGHGLXnK1fQQRiliAJQDiKGbRZMATgHM5tfpuv3HAoflEoAYsQmkACqWCnIAZkwqLFAatKERUTEeAL4A3Dg4oJCwEEoAXnJUUACeAA5yIGHwyCgAdHAk5BiaAO424Ha0qUA

Reply:
That's right, we need either `typeof` or `import typeof` at the top.

There are also helpers like `$Keys<Foo>` for dealing with enums and maps

Reply:
Done